xbmc/interfaces/json-rpc/PVROperations.cpp

```diff
@@ -138,11 +138,18 @@ JSONRPC_STATUS CPVROperations::GetChannelDetails(const std::string &method, ITra
   if (!channelGroupContainer)
     return FailedToExecute;
 
-  std::shared_ptr<CPVRChannel> channel = channelGroupContainer->GetChannelById((int)parameterObject["channelid"].asInteger());
+  std::shared_ptr<CPVRChannel> channel = channelGroupContainer->GetChannelById(
+      static_cast<int>(parameterObject["channelid"].asInteger()));
```

Comment:
This is the only part where this doesn't feel right. How do we know that we are retrieving the correct group member?

Probably not an issue if the channelid is group specific but I'm not entirely sure that is the case.

Reply:
The new code mimics exactly the logic as it was before. It obtains the member of the currently selected channel group or if the requested channel is not member of the selected group, from the all channels group. So, if later on the channel number should be queried from the created item, the exact number as before this PR should be returned.

The 'magic' happens in CServiceBroker::GetPVRManager().GUIActions()->GetChannelGroupMember() called some lines below.

Reply:
Cool, OK. Then all good.